src/cometh/rental-protocol/rental-offer-created-handler.ts

```diff
@@ -0,0 +1,65 @@
+import { BigInt } from '@graphprotocol/graph-ts'
+import { RentalOffer } from '../../../generated/schema'
+import { COMETHSPACESHIP } from '../../utils/constants'
+import { RentalOfferCreated } from '../../../generated/ComethRentalProtocol/ComethRentalProtocol'
+import { loadNfts } from '../../utils/misc'
+import { log } from '@graphprotocol/graph-ts'
+import { SPACESHIP_ADDRESS } from '../../utils/addresses'
+/**
+ *    event RentalOfferCreated(
+ *       uint256 indexed nonce,
+ *       address indexed maker,
+ *       address taker,
+ *       NFT[] nfts,
+ *       address feeToken,
+ *       uint256 feeAmount,
+ *       uint256 deadline
+ *   );
+ */
+export function handleRentalOfferCreated(event: RentalOfferCreated): void {
+  // filter out nfts that are not spaceships
+  const filteredNfts = event.params.nfts.filter((nft) => nft.token.toHexString() == SPACESHIP_ADDRESS)
+
+  const filteredTokenIds = filteredNfts.map<BigInt>((nft) => nft.tokenId)
+
+  if (filteredTokenIds.length == 0) {
+    log.debug('[handleRentalOfferCreated] No spaceship nfts with valid address found in rental offer, tx: {}', [
+      event.transaction.hash.toHex(),
+    ])
+    return
+  }
+
+  const nfts = loadNfts(filteredTokenIds, COMETHSPACESHIP)
+
+  if (!nfts.length) {
+    log.debug('[handleRentalOfferCreated] No valid spaceship nfts with valid id found in rental offer, tx: {}', [
+      event.transaction.hash.toHex(),
+    ])
+    return
+  }
+
+  // create rental offer
+  const rentalOfferId = `${event.params.maker.toHexString()}-${event.params.nonce}`
+  const rentalOffer = new RentalOffer(rentalOfferId)
+  rentalOffer.nfts = nfts.map<string>((nft) => nft.id)
```

Comment:
We filter out some NFTs but include them back here. Should we just put the filtered NFTs?

Reply:
Actually no, filteredTokenIds are the ids of only spaceships. But since the cometh rental protocol allow non-existent spaceships ids, the nfts contain only the ones existent. So it's not the same thing.

I will rename the variables for a better understanding